ComponentKit/Core/CKBuildComponent.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <ComponentKit/CKComponentScopeTypes.h>
+#import <ComponentKit/CKComponentBoundsAnimation.h>
+
+@protocol CKScopedComponent;
+
+@class CKComponentScopeRoot;
+@class CKComponent;
+
+struct CKBuildComponentResult {
+  CKComponent *component;
+  CKComponentScopeRoot *scopeRoot;
+  CKComponentBoundsAnimation boundsAnimation;
+};
+
+CKBuildComponentResult CKBuildComponent(CKComponentScopeRoot *previousRoot,
+                                        const CKComponentStateUpdateMap &stateUpdates,
+                                        CKComponent *(^function)(void));
```

Comment:
I know you are just moving code around, but can we give this a better name? `creationBlock`, maybe?